plugins/verifier/cosign/cosign.go

```diff
@@ -50,9 +52,13 @@ type StoreConfig struct {
 	AuthProvider string `json:"auth-provider,omitempty"`
 }
 
+type StoreWrapperConfig struct {
+	StoreConfig StoreConfig `json:"store"`
+}
+
 type PluginInputConfig struct {
-	Config      PluginConfig `json:"config"`
-	StoreConfig StoreConfig  `json:"storeConfig"`
+	Config             PluginConfig       `json:"config"`
+	StoreWrapperConfig StoreWrapperConfig `json:"storeConfig"`
```

Comment:
good question! turns out that all this time the Store config passed to the cosign plugin was not being deserialized correctly since the structs were nested one more layer. We never caught it because there were only 2 fields in that in any case `useHttp` and `cosignEnabled`. Neither of which were used until I enabled useHttp support here.

Reply:
Gotcha, good catch!